src/Symfony/Component/Notifier/Bridge/Mercure/MercureTransportFactory.php

```diff
@@ -45,6 +46,10 @@ public function create(Dsn $dsn): TransportInterface
             throw new UnsupportedSchemeException($dsn, 'mercure', $this->getSupportedSchemes());
         }
 
+        if (!class_exists(MercureBundle::class)) {
+            throw new LogicException('The MercureBundle is not registered in your application. Try running "composer require symfony/mercure-bundle".');
+        }
+
         $publisherId = $dsn->getHost();
         if (!$this->publisherLocator->has($publisherId)) {
```

Comment:
Maybe we should add the check here:

```php
if (!class_exists(MercureBundle::class && 0 === \count($this->publisherLocator->getProvidedServices()))) {
    throw new LogicException(sprintf('"%s" not found. Did you forget to install the MercureBundle? Try running "composer require symfony/mercure-bundle".', $publisherId));
} else {
    throw new LogicException(sprintf('"%s" not found. Did you mean one of: %s?', $publisherId, implode(', ', array_keys($this->publisherLocator->getProvidedServices()))));
}
```

What do you think?

Reply:
Sounds good!

Reply:
Or we can check as well the presence of `FrameworkExtension`? Somthing like:
```php
        if (!$this->publisherLocator->has($publisherId)) {
            if (!class_exists(MercureBundle::class) && class_exists(FrameworkExtension::class)) {
                throw new LogicException('The MercureBundle is not registered in your application. Try running "composer require symfony/mercure-bundle".');
            }
            
            if (0 === \count($this->publisherLocator->getProvidedServices())) {
                throw new LogicException('No publisher were found. $publisherLocator is empty.');
            }

            throw new LogicException(sprintf('"%s" not found. Did you mean one of: %s?', $publisherId, implode(', ', array_keys($this->publisherLocator->getProvidedServices()))));
        }

```

What do you think?